clients/fossid-webapp/src/test/kotlin/FossIdClientTestNewProject.kt

```diff
@@ -0,0 +1,224 @@
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration
+
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.collections.shouldBeEmpty
+import io.kotest.matchers.maps.shouldContain
+import io.kotest.matchers.nulls.shouldBeNull
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldContain
+
+import okhttp3.OkHttpClient
+
+import org.ossreviewtoolkit.fossid.FossIdRestService
+import org.ossreviewtoolkit.fossid.api.Scan
+import org.ossreviewtoolkit.fossid.api.identification.ignored.IgnoredFile
+import org.ossreviewtoolkit.fossid.api.identification.markedAsIdentified.MarkedAsIdentifiedFile
+import org.ossreviewtoolkit.fossid.api.status.DownloadStatus
+import org.ossreviewtoolkit.fossid.api.status.ScanState
+import org.ossreviewtoolkit.fossid.checkDownloadStatus
+import org.ossreviewtoolkit.fossid.checkResponse
+import org.ossreviewtoolkit.fossid.checkScanStatus
+import org.ossreviewtoolkit.fossid.createProject
+import org.ossreviewtoolkit.fossid.createScan
+import org.ossreviewtoolkit.fossid.downloadFromGit
+import org.ossreviewtoolkit.fossid.getProject
+import org.ossreviewtoolkit.fossid.listIdentifiedFiles
+import org.ossreviewtoolkit.fossid.listIgnoredFiles
+import org.ossreviewtoolkit.fossid.listMarkedAsIdentifiedFiles
+import org.ossreviewtoolkit.fossid.listScanResults
+import org.ossreviewtoolkit.fossid.listScansForProject
+import org.ossreviewtoolkit.fossid.runScan
+import org.ossreviewtoolkit.fossid.toList
+import org.ossreviewtoolkit.utils.test.shouldNotBeNull
+
+import retrofit2.Retrofit
+import retrofit2.converter.jackson.JacksonConverterFactory
+
+/**
+ * This client test creates a new project, triggers the download and the scan and gets the scan results
+ */
+class FossIdClientTestNewProject : StringSpec({
+    lateinit var wiremock: WireMockServer
+    lateinit var service: FossIdRestService
+
+    beforeSpec {
+        wiremock = WireMockServer(
+            WireMockConfiguration.options()
+                .dynamicPort()
+                .usingFilesUnderDirectory("src/test/assets/test1/")
+        )
+        wiremock.start()
+        WireMock.configureFor(wiremock.port())
+
+        val client = OkHttpClient.Builder()
+            .build()
+
+        val mapper = JsonMapper()
+            .registerKotlinModule()
+            .setPropertyNamingStrategy(PropertyNamingStrategies.SNAKE_CASE)
+        val retrofit = Retrofit.Builder()
+            .client(client)
+            .baseUrl("http://localhost:" + wiremock.port())
+            .addConverterFactory(JacksonConverterFactory.create(mapper))
+            .build()
+
+        service = retrofit.create(FossIdRestService::class.java)
+    }
+
+    afterSpec {
+        wiremock.stop()
+    }
+
+    beforeTest {
+        wiremock.resetAll()
+    }
+
+    val projectCode = "semver4j"
```

Comment:
How about making this a top-level `private` `const`?

Reply:
Done, also for `scanCode`, but they cannot be `const` because we are not in the context of a class.
These are local variables in the lambda expression passed to StringSpec's constructor.

Reply:
What I mean was to do

```
const val PROJECT_CODE = "semver4j"
const val SCAN_CODE = "${PROJECT_CODE}_20201203_090342"
```

on the top *file* level, i.e. on line 63 / 64, so we *can* actually make these `const`.